tests/python/gpu/test_contrib_amp.py

```diff
@@ -0,0 +1,255 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import sys
+import mxnet as mx
+import numpy as np
+import warnings
+import collections
+import ctypes
+import mxnet.contrib.amp as amp
+from nose.tools import assert_raises
+from mxnet.test_utils import set_default_context, download_model, compare_symbol_structure
+from mxnet.gluon.model_zoo.vision import get_model
+from mxnet.gluon import SymbolBlock
+from mxnet.contrib.amp import amp
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.insert(0, os.path.join(curr_path, '../unittest'))
+from common import with_seed, teardown
+
+def test_amp_coverage():
+    conditional = [item[0] for item in amp.lists.symbol.CONDITIONAL_FP32_FUNCS]
+
+    # Check for duplicates
+    for a in [amp.lists.symbol.FP16_FUNCS,
+          amp.lists.symbol.FP16_FP32_FUNCS,
+          amp.lists.symbol.FP32_FUNCS,
+          amp.lists.symbol.WIDEST_TYPE_CASTS,
+          conditional]:
+        ret = [item for item, count in collections.Counter(a).items() if count > 1]
+        assert ret == [], "Elements " + str(ret) + " are duplicated in the AMP lists."
+
+    t = []
+    for a in [amp.lists.symbol.FP16_FUNCS,
+              amp.lists.symbol.FP16_FP32_FUNCS,
+              amp.lists.symbol.FP32_FUNCS,
+              amp.lists.symbol.WIDEST_TYPE_CASTS,
+              conditional]:
+        t += a
+    ret = [item for item, count in collections.Counter(t).items() if count > 1]
+    assert ret == [], "Elements " + str(ret) + " exist in more than 1 AMP list."
+
+    # Check the coverage
+    py_str = lambda x: x.decode('utf-8')
+
+    plist = ctypes.POINTER(ctypes.c_char_p)()
+    size = ctypes.c_uint()
+
+    mx.base._LIB.MXListAllOpNames(ctypes.byref(size),
+                                     ctypes.byref(plist))
+    op_names = []
+    for i in range(size.value):
+        s = py_str(plist[i])
+        if not s.startswith("_backward") \
+           and not s.startswith("_contrib_backward_"):
+            op_names.append(s)
+
+    ret1 = set(op_names) - set(t)
+
+    if ret1 != set():
+        warnings.warn("Operators " + str(ret1) + " do not exist in AMP lists (in "
+                       "python/mxnet/contrib/amp/lists/symbol.py) - please add them. "
+                       """Please follow these guidelines for choosing a proper list:
+                       - if your operator is not to be used in a computational graph
+                         (e.g. image manipulation operators, optimizers) or does not have
+                         inputs, put it in FP16_FP32_FUNCS list,
+                       - if your operator requires FP32 inputs or is not safe to use with lower
+                         precision, put it in FP32_FUNCS list,
```

Comment:
Based on the discussion on the dev list didnt we decide not to have an explicit list for FP32_FUNCS? And instead assume anything not specified would go in this category? Otherwise this will be a maintenance nightmare and make adding new operators confusing.

Reply:
i think i just moved this to a different file. @ptrendx  owns this part and can add more info here.